src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/transport/action/removepolicy/TransportRemovePolicyAction.kt

```diff
@@ -55,7 +56,9 @@ class TransportRemovePolicyAction @Inject constructor(
         RemovePolicyAction.NAME, transportService, actionFilters, ::RemovePolicyRequest
 ) {
     override fun doExecute(task: Task, request: RemovePolicyRequest, listener: ActionListener<ISMStatusResponse>) {
-        RemovePolicyHandler(client, listener, request).start()
+        client.threadPool().threadContext.stashContext().use {
```

Comment:
I see that the `TransportAddPolicyAction` is retrieving user and passing it to the `AddPolicyHandler` but we don't seem to be adding user context here

Reply:
when we retrieve user, we are trying to save it to managed-index. So we only retrieve user for addPolicy, changePolicy, indexPolicy(save to ismTemplate field). Other actions we don't do that.